lib/gps.js

```diff
@@ -0,0 +1,418 @@
+  chip = opts.chip;
+
+  // If a breakout is defined check for receiver and chip
+  if (Breakouts[breakout]) {
+    if (!receiver && Breakouts[breakout].receiver) {
+      receiver = Breakouts[breakout].receiver.value;
+    }
+
+    if (!chip && Breakouts[breakout].chip) {
+      chip = Breakouts[breakout].chip.value;
+    }
+  }
+
+  // If a receiver was defined or derived but chip was not
+  if (receiver && !chip && Receivers[receiver].chip) {
+    chip = Receivers[receiver].chip.value;
+  }
+
+  // Default case
+  if (!chip) {
+    chip = "DEFAULT";
+  }
+
+  // Allow users to pass in custom chip types
+  chip = typeof chip === "string" ?
+    Chips[chip] : opts.chip;
+
+  // Allow users to pass in custom receiver types
+  receiver = typeof receiver === "string" ?
+    Receivers[receiver] : opts.receiver;
+
+  // Chip decorates the instance
+  Object.defineProperties(this, chip);
+
+  // Receiver decorates this instance
+  if (receiver) {
+    Object.defineProperties(this, receiver);
+  }
+
+  // breakout decorates the instance
+  if (opts.breakout) {
+    breakout = typeof opts.breakout === "string" ?
+      Breakouts[opts.breakout] : opts.breakout;
+
+    Board.Controller.call(this, breakout, opts);
+  }
+
+  // If necessary set default property values
+  this.precision = opts.precision || 9;
+  this.baud = opts.baud || breakout.baud || chip.baud;
+
+  // Create a "state" entry for privately
+  // storing the state of the instance
+  state = {
+    sat: {},
+    pos: {},
+    nav: {},
+    frequency: 1000,
+    lowPowerMode: false
+  };
+
+  priv.set(this, state);
+
+  // Getters for private state values
+  Object.defineProperties(this, {
+    pos: {
+      get: function() {
+        return state.pos;
+      }
+    },
+    sat: {
+      get: function() {
+        return state.sat;
+      }
+    },
+    nav: {
+      get: function() {
+        return state.nav;
+      }
+    },
+    time: {
+      get: function() {
+        return state.time;
+      }
+    }
+  });
+
+  if (this.initialize) {
+    this.initialize(opts);
+  }
+
+}
+
+util.inherits(GPS, events.EventEmitter);
+
+/*
+ * Default intialization for serial GPS
+ */
+GPS.prototype.initialize = function(opts) {
+
+  var state = priv.get(this);
+  state.portId = opts.serialPort || opts.portId || opts.port || opts.bus || this.io.SERIAL_PORT_IDs.SW_SERIAL0 || this.io.SERIAL_PORT_IDs[0];
+
+  // Set the pin modes
+  ["tx", "rx"].forEach(function(pin) {
+    if (this.pins[pin]) {
+      this.io.pinMode(this.pins[pin], this.io.MODES.SERIAL);
+    }
+  }, this);
+
+  if (this.pins.onOff) {
+    this.io.pinMode(this.pins.onOff, this.io.MODES.OUTPUT);
+    this.onOff = new Pin(this.pins.onOff);
+  }
+
+  this.io.serialConfig({
+    portId: state.portId,
+    baud: this.baud,
+    rxPin: this.pins.rx,
+    txPin: this.pins.tx
+  });
+
+  if (this.configure) {
+    this.configure(function() {
+      this.listen();
+      if (opts.freq) {
+        this.frequency(opts.freq);
```

Comment:
I get what you're saying re: changing frequency to an accessor property and can make that change. 

Won't hertz be the inverse on frequency? (1hz = 1000ms frequency, 1000hz = 1ms frequency)